src/dos/dos.cpp

```diff
@@ -148,7 +150,52 @@ static uint16_t DOS_GetAmount(void) {
 #ifndef DOSBOX_CPU_H
 #include "cpu.h"
 #endif
-static inline void modify_cycles(Bits value) {
+
+// Taken from Dosbox-X
+int hdd_data_rate = 0;
+int fdd_data_rate = 0;
+
+void DOS_SetDataRate(int rate, DiskType type)
+{
+	if (type == DiskType::Floppy) {
+		hdd_data_rate = rate * BytesPerKilobyte;
+	} else {
+		fdd_data_rate = rate * BytesPerKilobyte;
+	}
```

Comment:
We use `switch` statements when using enums whenever possible. That will force you handle all cases, and in this case you'd need to explain it in a comment why is the `CdRom` type ignored.

This is a good example. The default branch with `assertm` is needed to satisfy the static analysers we're using.

https://github.com/dosbox-staging/dosbox-staging/blob/94ff0ca45ea88b61eeada4944c953da390599ff1/src/gui/shader_manager.cpp#L573-L598